COMMANDS.org

```diff
@@ -0,0 +1,117 @@
+#+TITLE: Nix Command Reference
+#+AUTHOR: github.com/Pitta
+
+A reference to ~nix~ commands found to be helpful in day to day use.
+
+* Package Management
+
+Coming from package managers like ~brew~, ~port~, ~apt~, etc., ~nix~ can /seem/ daunting/impenetrable. These section should help demystify how to use ~nix~ as a drop in replacement for how most folks approach using the aforementioned tools.
+
+- [[#-searching-for-packages][Searching for Packages]]
+- [[#-imperitive-package-management][Imperitive Package Management]]
+  - how most users manage packages, typically by invoking an install command with the package manager of choice.
+- [[#-declaritive-package-management][Declaritive Package Management]]
+  - A more portable, version controll-able approach that declares the desired end state. Great for using the same configuration across multiple machines.
+- [[#-temporary-packages][Temporary Packages]]
+  - Excellent for testing new versions of packages, or entirely new tools, without comitting to system level changes that could impact other parts of the system.
+
+** 🔍 Searching for Pacakges
+
+#+begin_quote
+/Question: I want to install pacakges, but I'm not sure what is avalible/
+#+end_quote
+
+Search for packages via:
+- The web UI @ https://search.nixos.org/
+- Using the cli:
+  #+begin_src bash
+  nix-env -qaP 'some_package' 
+  #+end_src
+
+  - ~-q~: Stands for "query". Used to query the Nix store for /installed/ pacakges
+  - ~-a~: Stands for "avalible". Used to search for /avalible packages/ to be installed
+  - ~-P~: Includes the attribute path of the package. This result is often used with the ~-A~ flag during installs to more precisely target the desired package.
+
+** 🪛 Imperative Package Management
+
+#+begin_quote
+/Question: I'm used to tools like brew - how do I use nix in a similar way?/
+#+end_quote
+
+If your workflow has involved...
+- ~brew install some_package~
+- ~port install some_package~
+- ~api-get install some_package~
+
+...and are making the move to ~nix~, then you are in the right place! ~nix~ can be used as a drop in replacement for these tools on most linux/unix based systems.
+
+*** Install a package
+
+Once the desired package has been identified, use ~nix-env~ to install the package globally...
+
+#+begin_src bash
+  nix-env -i some_package
```

Comment:
Since this is meant to be more of a "make it easy" guide, I would change this to:

```
nix-env --install some_package
```

Because `--install` is easy to remember, it's also the action somebody wants to take so it is easier to remember